crates/bevy_scene/src/scene_spawner.rs

```diff
@@ -50,27 +52,50 @@ pub struct SceneSpawner {
     scenes_with_parent: Vec<(InstanceId, Entity)>,
 }
 
+/// Errors that can occur when spawning a scene.
 #[derive(Error, Debug)]
 pub enum SceneSpawnError {
+    /// Scene contains an unregistered component type.
     #[error("scene contains the unregistered component `{type_name}`. consider adding `#[reflect(Component)]` to your type")]
-    UnregisteredComponent { type_name: String },
+    UnregisteredComponent {
+        /// Name of the unregistered component type.
+        type_name: String,
+    },
+    /// Scene contains an unregistered resource type.
     #[error("scene contains the unregistered resource `{type_name}`. consider adding `#[reflect(Resource)]` to your type")]
-    UnregisteredResource { type_name: String },
+    UnregisteredResource {
+        /// Name of the unregistered resource type.
+        type_name: String,
+    },
+    /// Scene contains an unregistered type.
     #[error("scene contains the unregistered type `{type_name}`. consider registering the type using `app.register_type::<T>()`")]
-    UnregisteredType { type_name: String },
+    UnregisteredType {
+        /// Name of the unregistered type.
+        type_name: String,
+    },
+    /// Dynamic scene with the given id does not exist.
     #[error("scene does not exist")]
-    NonExistentScene { id: AssetId<DynamicScene> },
+    NonExistentScene {
+        /// Id of the non-existent dynamic scene.
+        id: AssetId<DynamicScene>,
+    },
+    /// Scene with the given id does not exist.
     #[error("scene does not exist")]
-    NonExistentRealScene { id: AssetId<Scene> },
+    NonExistentRealScene {
+        /// Id of the non-existent scene.
+        id: AssetId<Scene>,
+    },
```

Comment:
"Name of the unregistered FOOBAR type" doesn't sound right. Could those be replaced by "Type of the unregistered FOOBAR"?

Reply:
Sounds better indeed, what about `type_name` on `UnregisteredType`? Keep the same docstring, or something like "The unregistered type"?

Reply:
The idea is that since the field name is `type_name: String`, it would be easy for a reader to infer that it's indeed the name of the type that is being stored, without further clarification. But "Type name of the unregistered FOOBAR" works as well.